resources/sdk/pureclouddotnet/extensions/Extensions/Notifications/NotificationHandler.cs

```diff
@@ -72,10 +75,27 @@ public NotificationHandler(Channel channel, string proxyURI, string proxyUsernam
         {
             Channel = channel;
             ConnectSocket(Channel.ConnectUri, proxyURI, proxyUsername, proxyPassword);
+            SubscribeToSystemEvents();
         }
 
         /// <summary>
-        /// Adds a subsciption to the specified topic. Events received on this topic will be cast to the given type.
+        /// Adds a handler for the v2.system.socket_closing message in order to reconnect to the websocket when the message is received
+        /// </summary>
+        private void SubscribeToSystemEvents()
+        {
+            AddHandlerNoSubscribe("v2.system.socket_closing", typeof(SystemMessageSystemMessage));
+            NotificationReceived += (data) =>
+            {
+                if (data.GetType() == typeof (NotificationData<SystemMessageSystemMessage>))

```

Comment:
This condition should also explicitly check the topic name to filter out any other system topics the consumer may manually subscribe to.